pkg/validator/container.go

```diff
@@ -182,6 +184,54 @@ func (cv *ContainerValidation) validateImage(imageConf *conf.Images) {
 			cv.addSuccess(messages.ImageTagSuccess, category, id)
 		}
 	}
+
+	// Check the container image against the user-defined image registry whitelist and blacklist.
+	imageLists := [][]string{imageConf.Whitelist.Error, imageConf.Whitelist.Warning, imageConf.Blacklist.Error, imageConf.Blacklist.Warning}
+	hasWhitelistCheck := false
+	hasWhitelistFailure := false
+	whitelistCheckID := "imageWhitelist"
+	hasBlacklistCheck := false
+	hasBlacklistFailure := false
+	blacklistCheckID := "imageBlacklist"
+
+	for _, imageList := range imageLists {
+		var severity config.Severity
+		if &imageList == &imageConf.Whitelist.Error {
+			severity = config.SeverityError
+		} else {
+			severity = config.SeverityWarning
+		}
+		for _, imagePattern := range imageList {
+			hasWhitelistCheck = true
+			if !regexp.MustCompile(imagePattern).MatchString(cv.Container.Image) {
+				hasWhitelistFailure = true
+				cv.addFailure(messages.ImageRegistryFailure, severity, category, whitelistCheckID)
+			}
+		}
+
+	}
+	if hasBlacklistCheck && hasBlacklistFailure == false {
+		cv.addSuccess(messages.ImageRegistrySuccess, category, blacklistCheckID)
+	}
+	for _, imageList := range imageLists {
+		var severity config.Severity
+		if &imageList == &imageConf.Blacklist.Error {
```

Comment:
There are a few instances of `blacklist` in this block that need to become `whitelist`

Reply:
Updated.